examples/AspNetCore/Startup.cs

```diff
@@ -94,10 +95,9 @@ public void ConfigureServices(IServiceCollection services)
                         .SetResourceBuilder(ResourceBuilder.CreateDefault().AddService(this.Configuration.GetValue<string>("Otlp:ServiceName")))
                         .AddAspNetCoreInstrumentation()
                         .AddHttpClientInstrumentation()
-                        .AddOtlpExporter(otlpOptions =>
-                        {
-                            otlpOptions.Endpoint = new Uri(this.Configuration.GetValue<string>("Otlp:Endpoint"));
-                        }));
+                        .AddOtlpExporter(new OltpTraceExporterOptionsBuilder()
+                            .ConfigureEndpoint(new Uri(this.Configuration.GetValue<string>("Otlp:Endpoint")))
+                            .ConfigureActivityProcessor(o => o.ExportProcessorType = ExportProcessorType.Batch)));
```

Comment:
I do like the ergonomics of the builder pattern you're proposing. If it's only used with the `Add*Exporter()` extension methods then I think it may better help people understand that they're doing more to their pipeline than just adding an exporter - that is, in this example it is pretty apparent that I'm configuring the exporter's endpoint and an activity processor.

What I don't like today about the `OtlpExporterOptions` (and Jaeger/Zipkin/Console/etc) class is that it encompasses options about both the exporter plus other things - e.g. activity processors and metric readers (#2552). Furthermore, the `AddOtlpExporter` extension method and the OTLP trace exporter constructor accepts an `OtlpExporterOptions` instance. In the former the processor/metric reader options are meaningful and the latter it is not.

I still like the idea of having the options classes only hold options relevant to the exporters themselves - and move the non-exporter related options out to their own options classes. This is part of what I'm doing on #2717. I think this improves the use case where I'm not using the extension method to configure my pipeline.

What are your thoughts about separating the exporter options from the processor/metric reader options like I have on #2717 and then have the `OltpTraceExporterOptionsBuilder` have a handle on both sets of options?